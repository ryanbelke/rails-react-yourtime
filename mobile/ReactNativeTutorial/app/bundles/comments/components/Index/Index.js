// @flow
import React from 'react';
import { View } from 'react-native';
import List from './List/List';
import Footer from './Footer/Footer';
import type { IndexPropsType } from '../../containers/Index';

import styles from './IndexStyle';

type PropsType = IndexPropsType;

const Index = (props: PropsType) => (
  <View style={styles.container}>
    <List {...props} />
    <Footer />
  </View>
);

export default Index;
