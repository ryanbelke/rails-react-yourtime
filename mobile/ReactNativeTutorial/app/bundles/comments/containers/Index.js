import _ from 'lodash/fp';
import withIndexProps from '../hocs/withIndexProps';
import withInitialAction from '../../common/hocs/withInitialAction';
import Index from '../components/Index/Index';

export default _.compose(
  withIndexProps,
  withInitialAction(_.get(['actions', 'fetch']))
)(Index);
