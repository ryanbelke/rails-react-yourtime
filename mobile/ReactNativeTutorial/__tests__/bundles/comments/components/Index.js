import React from 'react';
import Index from 'ReactNativeTutorial/app/bundles/comments/components/Index/Index';

import renderer from 'react-test-renderer';

const actions = {
    fetch: jest.fn(),
    updateForm: jest.fn(),
    createComment: jest.fn(),
};

describe('Index', () => {
  it('renders correctly when loaded', () => {
    const props = {
      comments: [
        { id: 1, author: 'Alexey', text: 'Just some random comment' },
        { id: 2, author: 'Justin', text: 'Another random comment' },
      ],
      meta: {
        loading: false,
      },
      actions,
    };
    const tree = renderer.create(
      <Index {...props} />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const props = {
      meta: {
        loading: true,
      },
      actions,
    };
    const tree = renderer.create(
      <Index {...props} />
    );
    expect(tree).toMatchSnapshot();
  });
});

