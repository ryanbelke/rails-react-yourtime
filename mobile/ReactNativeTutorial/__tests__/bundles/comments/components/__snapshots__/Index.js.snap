exports[`Index renders correctly when loaded 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 2,
      }
    }
    enableEmptySections={true}
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    refreshControl={
      <RefreshControl
        onRefresh={[Function]}
        refreshing={false}
        tintColor="#4641B5" />
    }
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}>
    <View
      style={
        Object {
          "backgroundColor": "#F0F0F0",
          "elevation": 5,
          "marginHorizontal": 20,
          "marginVertical": 10,
          "minHeight": 80,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 5,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        Alexey
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#555555",
            "marginBottom": 10,
            "marginHorizontal": 10,
          }
        }>
        Just some random comment
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F0F0F0",
          "elevation": 5,
          "marginHorizontal": 20,
          "marginVertical": 10,
          "minHeight": 80,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 5,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        Justin
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#555555",
            "marginBottom": 10,
            "marginHorizontal": 10,
          }
        }>
        Another random comment
      </Text>
    </View>
  </ScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EFEFF2",
        "borderTopColor": "#D2D2D2",
        "borderTopWidth": 1,
        "height": 70,
        "justifyContent": "center",
      }
    }>
    <View
      onPress={undefined}
      style={
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 28,
            "height": 52,
            "width": 52,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#f50",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="#f50">
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "Material Icons",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
          ]
        }>
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Index renders correctly when loading 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    enableEmptySections={true}
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    refreshControl={
      <RefreshControl
        onRefresh={[Function]}
        refreshing={true}
        tintColor="#4641B5" />
    }
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []} />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EFEFF2",
        "borderTopColor": "#D2D2D2",
        "borderTopWidth": 1,
        "height": 70,
        "justifyContent": "center",
      }
    }>
    <View
      onPress={undefined}
      style={
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 28,
            "height": 52,
            "width": 52,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#f50",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="#f50">
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "Material Icons",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
          ]
        }>
        
      </Text>
    </View>
  </View>
</View>
`;
